packages/pwa-buildpack/lib/WebpackTools/plugins/LocalizationPlugin.js

```diff
@@ -0,0 +1,262 @@
+const debug = require('../../util/debug').makeFileLogger(__filename);
+const path = require('path');
+const walk = require('../../util/klaw-bound-fs');
+const merge = require('merge');
+const InjectPlugin = require('webpack-inject-plugin').default;
+
+const i18nDir = 'i18n';
+const localeFileNameRegex = /([a-z]{2}_[A-Z]{2})\.json$/;
+
+/**
+ * Localization Plugin to collect all translations from NPM modules and create single translation file for each locale
+ *
+ * Options:
+ *  - context: string provide the context that the build is running within
+ *  - dirs[]: array of directories to search for i18n/*.csv files
+ *  - virtualModules: instance of VirtualModulesPlugin to create virtual modules during the build
+ */
+class LocalizationPlugin {
+    /**
+     * @param {object} opts
+     */
+    constructor(opts) {
+        this.opts = opts;
+    }
+
+    apply(compiler) {
+        this.compiler = compiler;
+        this.injectLocalizationLoader();
+    }
+
+    injectLocalizationLoader() {
+        debug(
+            'applying InjectPlugin to create global __fetchLocaleData__ for localization import'
+        );
+        new InjectPlugin(() => this.buildFetchModule()).apply(this.compiler);
+    }
+
+    async buildFetchModule() {
+        const { inputFileSystem, hooks } = this.compiler;
+        const { dirs, context, virtualModules } = this.opts;
+
+        // Iterate through every module which declares the i18n specialFlag along with the context venia-concept dir
+        let locales = {};
+        for (const dir of dirs) {
+            const localeDir = path.join(dir, i18nDir);
+            try {
+                const stats = inputFileSystem.statSync(localeDir);
+
+                if (stats.isDirectory()) {
+                    const packageTranslations = await this.findTranslationFiles(
+                        inputFileSystem,
+                        localeDir
+                    );
+                    locales = this.mergeLocales(locales, packageTranslations);
```

Comment:
Does `mergeLocales` merge into the passed `current` ie `locales` object? Might not need the return here or in the function.

Reply:
Oops, writing PHP for so long makes you forget JS is pass by reference land. Good catch thank you :)